release/main.py

```diff
@@ -35,20 +39,69 @@ def tag_all(manifest, tag, delete_if_exists=False):
                 gtools.delete_tag(repo, tag)
         else:
             raise ValueError("%s tag already exists in these repositories %s" % (tag, repos_with_tag))
-    for repo, revision in manifest_contnet.items():
-        logger.info("Creating %s tag in repo: %s, revision: %s", tag, repo, revision)
-        gtools.create_tag(repo, revision, tag)
+    for repo, repo_info in manifest_contnet.items():
+        logger.info("Creating %s tag in repo: %s, revision: %s", tag, repo, repo_info['revision'])
+        gtools.create_tag(repo, repo_info['revision'], tag)
     logger.info("Done")
 
 
+def untag_all(manifest, tag):
+    '''Read the manifest file and delete the tag for each entry'''
+    logger = get_logger()
+    with open(manifest, 'r') as manifest_file:
+        manifest_contnet = yaml.safe_load(manifest_file)
+    logger.info("Deleting %s tag from all repos", tag)
+    gtools = gittools.GitApiUtils()
+    repos_with_tag = []
+    for repo in manifest_contnet.keys():
+        if gtools.tag_exists(repo, tag):
+            repos_with_tag.append(repo)
+    if repos_with_tag:
+        for repo in repos_with_tag:
+            gtools.delete_tag(repo, tag)
+
+
+def check_images_exists(manifest, tag):
```

Comment:
Notice that if we used the '--force' option than this validation is more complicated.
Its not enough to check that the tag exists. Need to verify that the SHA returned for the tag matches the SHA returned for the images with the git-hash tag.

Reply:
Thx, didn't think of that.